### PR TITLE
feat(view): ability to centralize node initially

### PR DESCRIPTION
**Pull request by kylo252**
This is only relevant for large nested folders, but I find it easier when the first selection is centered.

```help
    *nvim-tree.view.centralize_selection*
    Reposition the view so that the current node is initially centralized, see |zz|.
      Type: `boolean`, Default: `false`
```

Hopefully this is useful for others :smile:

### CODE REVIEW

File: lua/nvim-tree.lua
Comment (alex-courtis) [inline lua/nvim-tree.lua:352]: Does not obey `hijack_cursor = true`.

Putting this autocmd before `CursorMoved` does not fix this.

Please try changing this autocmd to be nested to see if you can make hijack_cursor work. If it's not possible I am happy to let it go.

File: lua/nvim-tree/view.lua
Comment (alex-courtis) [inline lua/nvim-tree/view.lua:398]: Let's not do this refactor. It does not appear to be necessary.

File: doc/nvim-tree-lua.txt
Comment (alex-courtis) [inline doc/nvim-tree-lua.txt:429]: Please clarify that this is on BufEnter.

Perhaps: "When entering nvim-tree, reposition the view so that the current node is moved to the center, see |zz|.

**Review by alex-courtis — CHANGES_REQUESTED** [review]
Looks good, works.

File: lua/nvim-tree/view.lua
Comment (kylo252) [reply lua/nvim-tree/view.lua:398]: > Let's not do this refactor. It does not appear to be necessary.

I mean it's not necessary in the sense that you can keep copying all the parameters by hand and hope you didn't forget any of them, or you can make it both explicit and obvious that this is the intended behavior. 

I'll put it back regardless, but I just wanted to point that out.

---

_P.s. notice how calling this merge function notified me that `height`/`weight` are initially missing.._

Comment (kyazdani42) [reply lua/nvim-tree/view.lua:398]: please rebase, i've fixed this issue when height is top/bottom

File: lua/nvim-tree.lua
Comment (kylo252) [reply lua/nvim-tree.lua:352]: I fixed it for now, but I do think that the problem originates from event used for `hijack_cursor`, example: try `:NvimTreeToggle` while having `hijack_cursor` to be set.  I don't think this would've been a problem, if the initial "legal" position is set when entering.

That said, I'm not sure it's worth the effort or the complexity to try to cover all the edge-cases that would originate from such a change.

**Review by kyazdani42 — APPROVED** [review]
looks good to me, waiting for alex's approval :)

**Review by alex-courtis — CHANGES_REQUESTED** [review]
Remove `place_cursor_on_node` and we are OK to merge.

**Review by alex-courtis — APPROVED** [review]

**Pull request merged**
